code/src/types/user.types.ts

```diff
@@ -23,3 +23,7 @@ export type TokenPayload = {
 };
 
 export type UserDetails = User & TokenPayload;
+
+export type UserId = {
+  userId: number;
+};
```

Comment:
also use this type for BaseUser. Like on the screenshot
![image](https://github.com/user-attachments/assets/3431014c-df81-43b6-99ba-5831c02f59e9)

Reply:
> also use this type for BaseUser. Like on the screenshot ![image](https://private-user-images.githubusercontent.com/118522791/348223225-3431014c-df81-43b6-99ba-5831c02f59e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3ODAxMzYsIm5iZiI6MTcyMDc3OTgzNiwicGF0aCI6Ii8xMTg1MjI3OTEvMzQ4MjIzMjI1LTM0MzEwMTRjLWRmODEtNDNiNi05OWJhLTU4MzFjMDJmNTllOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxMlQxMDIzNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjcyNWZlYWExMGY3MTY2ZmIwZGNmZmViODE2YzI4NGVhZmNkNmVhODBmMGZjYzM5NGFhYmIyNWRlMTdhZjJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NpZ8FZpPziuBLQqzwv2RKGARjwA5M0_yLcL9CfiraCE)

There will be an error
<img width="472" alt="image" src="https://github.com/user-attachments/assets/92493b48-2bb8-43ab-8252-300fa5053279">

Reply:
yes. Because you did UserId like an object. It's just a number . Make it like on the screenshot and everything will be okay. 
`export type UserId = number`.